js/ui/calendar/ui.calendar.base_view.js

```diff
@@ -253,36 +277,44 @@ const BaseView = Widget.inherit({
         return this._$table.find(`.${CALENDAR_CONTOURED_DATE_CLASS}`);
     },
 
-    _changeValue: function(cellDate) {
-        if(cellDate) {
-            const value = this.option('value');
-            const newValue = value ? new Date(value) : new Date();
-
-            newValue.setDate(cellDate.getDate());
-            newValue.setMonth(cellDate.getMonth());
-            newValue.setFullYear(cellDate.getFullYear());
-            newValue.setDate(cellDate.getDate());
+    _renderValue: function() {
+        if(!this.option('allowValueSelection')) {
+            return;
+        }
 
-            this.option('value', newValue);
-        } else {
-            this.option('value', null);
+        let value = this.option('value');
+        if(!Array.isArray(value)) {
+            value = [value];
         }
+
+        this._selectedCells?.forEach((cell) => { cell.removeClass(CALENDAR_SELECTED_DATE_CLASS); });
+
+        const selectedCells = value.map((value) => this._getCellByDate(value));
+
+        this._selectedCells = selectedCells;
+        selectedCells.forEach((cell) => { cell.addClass(CALENDAR_SELECTED_DATE_CLASS); });
```

Comment:
If `this._selectedCells` is array of jquery objects we shouldn't use iterator for adding class to each object. We may call this._selectedCells.addClass(CALENDAR_SELECTED_DATE_CLASS) only. Maybe I am missing something. Let me know if it's not acceptable here.

Reply:
Not sure I understand u correctly. We cannot call jQuery methods on js array, even if it is array of jQuery objects: https://codepen.io/zedwag/pen/rNZedGd?editors=0010

Reply:
There are troubles with this approach. Let's skip it.